arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class MapImageLayerTablesViewController: UIViewController {
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            // Assign the map to the MapView.
+            mapView.map = makeMap()
+            // Set the viewpoint.
+            mapView.setViewpoint(AGSViewpoint(latitude: 41.734152, longitude: -88.163718, scale: 2e5))
+            // Create a graphics overlay to show selected features and add it to the map view.
+            mapView.graphicsOverlays.add(selectedFeaturesOverlay)
+        }
+    }
+    
+    @IBOutlet var queryBarButtonItem: UIBarButtonItem!
+    
+    @IBAction func queryFeaturesActions() {
+        // Create an action sheet to display the various comments to choose from.
+        let alertController = UIAlertController(title: "Related Service Requests", message: "Select a comment to view related spatial features on the map.", preferredStyle: .actionSheet)
+        // Create an action for each comment.
+        commentsArray.forEach { feature in
+            // Extract the "comments" attribute as a string.
+            let commentsTitle = feature.attributes["comments"] as! String
+            // Create an action with the comments title.
+            let action = UIAlertAction(title: commentsTitle, style: .default) { [weak self] (_) in
```

Comment:
It's OK to use weak self; unowned self is also OK because the async UI always returns when the view controller exists.